datasets/noaa-cdr/noaa_cdr.py

```diff
@@ -0,0 +1,189 @@
+import logging
+import os.path
+from tempfile import TemporaryDirectory
+from typing import List, Union
+
+import stactools.noaa_cdr.constants
+import stactools.noaa_cdr.ocean_heat_content.stac
+import stactools.noaa_cdr.sea_ice_concentration.stac
+import stactools.noaa_cdr.sea_surface_temperature_optimum_interpolation.stac
+import stactools.noaa_cdr.sea_surface_temperature_whoi.stac
+import stactools.noaa_cdr.stac
+from pystac import Item, MediaType
+
+from pctasks.core.models.task import WaitTaskResult
+from pctasks.core.storage import StorageFactory
+from pctasks.dataset.collection import Collection
+
+logger = logging.getLogger(__name__)
+
+
+def cog_uri(name: str) -> str:
+    return f"blob://noaacdr/cogs/{name}"
+
+
+class OceanHeatContentCollection(Collection):
+    @classmethod
+    def create_item(
+        cls, asset_uri: str, storage_factory: StorageFactory
+    ) -> Union[List[Item], WaitTaskResult]:
+        directory = "/".join(asset_uri.split("/")[:-1])
+        asset_storage = storage_factory.get_storage(directory)
+        files = asset_storage.list_files(extensions=[".nc"])
+        cog_storage = storage_factory.get_storage(cog_uri("ocean-heat-content"))
+        cog_files = cog_storage.list_files(extensions=[".tif"])
+        cog_hrefs = [cog_storage.get_url(file) for file in cog_files]
+        with TemporaryDirectory() as temporary_directory:
+            local_netcdf_paths = []
+            for file in files:
+                path = os.path.join(temporary_directory, os.path.basename(file))
+                asset_storage.download_file(file, path)
+                local_netcdf_paths.append(path)
+            items = stactools.noaa_cdr.ocean_heat_content.stac.create_items(
+                hrefs=local_netcdf_paths,
+                directory=temporary_directory,
+                cog_hrefs=cog_hrefs,
+            )
+            # TODO add a derived-from link
```

Comment:
Are you able to finish off this TODO? It would be nice to have the derived from links, if possible. There's an example at https://github.com/microsoft/planetary-computer-tasks/blob/10464f7ab236a2fb5bc8856af18eef196c122a41/datasets/esa-cci-lc/esa_cci_lc.py#L42-L53.

Reply:
Does the relative url in `target` get normalized somehow downstream?

Reply:
I believe so, somewhere in our stac-fastapi deployment (https://github.com/microsoft/planetary-computer-apis/pull/113/files).

Reply:
Sweet, I'll yolo-trust that example then :-). Thanks!